src/components/Table/index.tsx

```diff
@@ -188,486 +485,287 @@ interface TableState<T> {
-
-		if (!key) {
-			return console.warn('onRowClick called on an element without a key set');
+		const indexOfColumn = this.state.allColumns.indexOf(column);
+		if (indexOfColumn < 0) {
+			return;
 		}
-		const row = this.getElementFromKey(key);
 
-		if (row) {
-			this.props.onRowClick(row, e);
-		}
+		const newColumn = getTagTableColumn(
+			getNewTagTableColumnState(tagKey),
+			this.props.tagField,
+		);
+		const columns = this.state.allColumns.slice();
+		columns.splice(indexOfColumn, 1, newColumn);
+		// trigger a refresh
+		this.setColumnsAndSave(columns);
 	};
 
-	public setPage = (change: any) => {
-		if (this.props.onPageChange) {
-			this.props.onPageChange(change);
+	protected onSort = (sort: TableSortOptions<T>) => {
+		if (this.props.sortingStateRestorationKey) {
+			saveSortingPreferences(this.props.sortingStateRestorationKey, sort);
+			this.setState({
+				sort,
+			});
 		}
 
-		this.setState({ page: change });
-	};
-
-	public resetPager = () => {
-		this.setPage(0);
-	};
-
-	public incrementPage = () => {
-		this.setPage(this.state.page + 1);
-	};
-
-	public decrementPage = () => {
-		this.setPage(this.state.page - 1);
+		if (this.props.onSort) {
+			this.props.onSort(sort);
+		}
 	};
 
 	public render() {
 		const {
 			columns,
-			data,
-			usePager,
-			itemsPerPage,
-			pagerPosition,
-			rowAnchorAttributes,
 			rowKey,
-			onCheck,
+			tagField,
+			columnStateRestorationKey,
+			loadColumnPreferences,
+			saveColumnPreferences,
+			sort: sortProp,
+			sortingStateRestorationKey,
+			innerRef,
+			enableCustomColumns,
+			onSort,
 			onRowClick,
-			getRowHref,
-			getRowClass,
 			...props
 		} = this.props;
 
-		const { page, sort } = this.state;
-		const items = data || [];
-		const totalItems = items.length;
-
-		const _itemsPerPage = itemsPerPage || 50;
-		const _pagerPosition = pagerPosition || 'top';
-
-		const lowerBound = usePager ? page * _itemsPerPage : 0;
-		const upperBound = usePager
-			? Math.min((page + 1) * _itemsPerPage, totalItems)
-			: totalItems;
-
-		const sortedData = this.sortData(items).slice(lowerBound, upperBound);
-
-		const shouldShowPaper = !!usePager && totalItems > 0;
+		const sort =
+			sortProp || (sortingStateRestorationKey ? this.state.sort : undefined);
 
 		return (
-			<>
-				{shouldShowPaper &&
-					(_pagerPosition === 'top' || _pagerPosition === 'both') && (
-						<Pager
-							totalItems={totalItems}
-							itemsPerPage={_itemsPerPage}
-							page={page}
-							nextPage={this.incrementPage}
-							prevPage={this.decrementPage}
-							mb={2}
-						/>
-					)}
-
-				<BaseTableWrapper>
-					<Base
+			<Container>
+				<Flex flexDirection="column" flex="1" width="100%">
+					<TypedTable<T>
+						enableCustomColumns={enableCustomColumns}
+						columns={this.state.visibleColumns}
+						rowKey={rowKey}
+						getRowHref={({ id }) => `${window.location.pathname}/${id}`}
+						ref={innerRef}
+						sort={sort}
+						onSort={this.onSort}
+						onRowClick={onRowClick}
```

Comment:
```suggestion
						columns={this.state.visibleColumns}
						ref={innerRef}
						sort={sort}
						onSort={this.onSort}
```
and just rely to the `{...props}` right below to wire them